packages/cloud_firestore/lib/src/field_path.dart

```diff
@@ -0,0 +1,11 @@
+// Copyright 2017, the Chromium project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+part of cloud_firestore;
+
+/// A [FieldPath] refers to a field in a document.
+class FieldPath {
+  /// The path to the document id, which can be used in queries.
+  static String get documentId => '__name__';
```

Comment:
This should return a `FieldPath` rather than a `String` to match the native API.

You can follow the pattern established for `FieldValue`: https://github.com/FirebaseExtended/flutterfire/blob/master/packages/cloud_firestore/lib/src/field_value.dart

Reply:
This is how I structured the pull request at first, however, I noticed that doing so would require an awful amount of changes, i.e. every query `String` parameter would need to be converted to `dynamic` and all the methods would need to convert the `FieldValue` to a `String` at some point. Another option would be to require the user to call `FieldValue.documentId.toString()`, but that is very strange usage and just a worse version of returning `String` right away. What do you think?

Reply:
This was how I structured my pull request at first, however, I thought that there were too many changes for this simple addition, but I can implement it.

Reply:
Thank you!